Userland/Libraries/LibCore/MimeData.cpp

```diff
@@ -70,33 +71,45 @@ void MimeData::set_text(const String& text)
     set_data("text/plain", text.to_byte_buffer());
 }
 
-String guess_mime_type_based_on_filename(const StringView& path)
+Optional<Core::MimeType> guess_mime_type_based_on_filename(const StringView& path)
 {
+    // TODO: This probably should be a JSON file soon...
     if (path.ends_with(".pbm", CaseSensitivity::CaseInsensitive))
-        return "image/x‑portable‑bitmap";
+        return Core::MimeType(Core::MimeType::m_type_image, "x‑portable‑bitmap");
     if (path.ends_with(".pgm", CaseSensitivity::CaseInsensitive))
-        return "image/x‑portable‑graymap";
+        return Core::MimeType(Core::MimeType::m_type_image, "x‑portable‑graymap");
     if (path.ends_with(".png", CaseSensitivity::CaseInsensitive))
-        return "image/png";
+        return Core::MimeType(Core::MimeType::m_type_image, "png");
     if (path.ends_with(".ppm", CaseSensitivity::CaseInsensitive))
-        return "image/x‑portable‑pixmap";
+        return Core::MimeType(Core::MimeType::m_type_image, "x‑portable‑pixmap");
     if (path.ends_with(".gif", CaseSensitivity::CaseInsensitive))
-        return "image/gif";
+        return Core::MimeType(Core::MimeType::m_type_image, "gif");
     if (path.ends_with(".bmp", CaseSensitivity::CaseInsensitive))
-        return "image/bmp";
+        return Core::MimeType(Core::MimeType::m_type_image, "bmp");
     if (path.ends_with(".jpg", CaseSensitivity::CaseInsensitive) || path.ends_with(".jpeg", CaseSensitivity::CaseInsensitive))
-        return "image/jpeg";
+        return Core::MimeType(Core::MimeType::m_type_image, "jpeg");
     if (path.ends_with(".svg", CaseSensitivity::CaseInsensitive))
-        return "image/svg+xml";
+        return Core::MimeType(Core::MimeType::m_type_image, "svg+xml");
     if (path.ends_with(".md", CaseSensitivity::CaseInsensitive))
-        return "text/markdown";
+        return Core::MimeType(Core::MimeType::m_type_text, "markdown");
     if (path.ends_with(".html", CaseSensitivity::CaseInsensitive) || path.ends_with(".htm", CaseSensitivity::CaseInsensitive))
-        return "text/html";
+        return Core::MimeType(Core::MimeType::m_type_text, "html");
     if (path.ends_with("/", CaseSensitivity::CaseInsensitive))
-        return "text/html";
+        return Core::MimeType(Core::MimeType::m_type_text, "html");
     if (path.ends_with(".csv", CaseSensitivity::CaseInsensitive))
-        return "text/csv";
-    return "text/plain";
+        return Core::MimeType(Core::MimeType::m_type_text, "csv");
```

Comment:
It feels rather odd to have `m_*` as part the public API of this thing
Maybe put that in a `Type` struct and just name it `text`?

Reply:
Agreed. `Core::MimeType::m_type_text` => `Core::MimeType::Type::text`